astropy/wcs/wcs.py

```diff
@@ -3093,11 +3093,7 @@ def slice(self, view, numpy_order=True):
                 if iview.step not in (None, 1):
                     crpix = self.wcs.crpix[wcs_index]
                     cdelt = self.wcs.cdelt[wcs_index]
-                    # equivalently (keep this comment so you can compare eqns):
-                    # wcs_new.wcs.crpix[wcs_index] =
-                    # (crpix - iview.start)*iview.step + 0.5 - iview.step/2.
-                    crp = ((crpix - iview.start - 1.)/iview.step
-                           + 0.5 + 1./iview.step/2.)
+                    crp = (crpix - 1 - iview.start) / iview.step + 1
```

Comment:
This implements the first version you discussed in #10864, right? This should be noted somewhere.

Reply:
I was thinking about re-binning vs sampling as well but I _thought/assumed_ that a `slice` with a parameter like `step` would indicate (or make `me` expect) _sampling_ as this is what essentially happens when you apply a slice to a `numpy` array.

With regard to the comment, I think it should talk about `stop` instead of `step`. I believe `stop` is currently ignored.

Reply:
`stop` seems to be accounted for here https://github.com/astropy/astropy/blob/ac313b3f17c19c734d8cd7e329470ef467ef1a71/astropy/wcs/wcs.py#L3090
But the note on `step` was directly taken from the docstring of the old function, which did not implement it.

It depends indeed on how you interpret the slicing operation, and how it is used _in praxi_. I merely guess the original code was intended to implement binning (which probably is more common when downscaling the number of pixels in an image).